Modernized

seripap opened the pull request:
Small caveat for this PR, since it's browserified, the min version is now 2.6kb. Probably not the best option to keep it small, but it works for my use case. Also should close #61

1. ES6-ified
2. Removed grunt
3. Switch testing suite to Jest
4. Using browserify

`npm run prepublish` handles building and minifying before publishing to npm

kaimallea commented:
You are a legend. 😮 But why does reducing the amount of code produce a bigger production file size?

kaimallea commented on README.md:
```diff
@@ -6,23 +6,17 @@
-## Why use isMobile?
+## Usage
 
-You probably shouldn't use this library unless you absolutely have to. In most cases, good [responsive design](https://en.wikipedia.org/wiki/Responsive_web_design) solves the problem of controlling how to
```
It might sound silly, but I would actually prefer to continue convincing people to avoid user agent sniffing unless absolutely necessary. 😃

seripap replied:
Yeah good catch- deleting everything in the readme file was an accident. I just reverted and rebased with only updated files

kaimallea commented on package.json:
```diff
@@ -18,11 +18,28 @@
+  "scripts": {
+    "test": "jest --coverage",
+    "dev": "jest --watch",
+    "prepublish": "rimraf dist && npm run build && npm run minify",
```
why do we need an npm package to delete a directory?

seripap replied:
🤷‍♂️. We don't actually

seripap commented:
Just updated and rebased;

1. Reverted the README back to the OG- just cleaned up the bottom half.
2. Re-added the original isMobile.min.js in the root as it's actually smaller than the browserfied/minified version
3. Removed rimraf in favor of `rm -rf`
4. Reverted src filename back to isMobile.js

To answer your question about the bigger production file; I believe it's because of the transpiling to support ES6 classes for older browsers. Unfortunately the only way around this is to write your own way of handling classes (as you did) or just go purely with functions.

kaimallea commented:
#77 modernized the tests, dependencies and build step. However, using ES6 in the source file and then bundling with browserify or webpack will introduce overhead that doubles the size of the generated file for browsers. :disappointed:
